xbmc/utils/test/TestURIUtils.cpp

```diff
@@ -620,3 +620,52 @@ TEST_F(TestURIUtils, ContainersEncodeHostnamePaths)
   EXPECT_EQ("rar://%2fpath%2fthing/my/archived/path",
             URIUtils::CreateArchivePath("rar", curl, "/my/archived/path").Get());
 }
+
+TEST_F(TestURIUtils, ServiceAvailabilityInTestEnvironment)
+{
+  using enum LanCheckMode;
+  {
+    EXPECT_FALSE(URIUtils::IsOnLAN("0.0.0.0", ANY_PRIVATE_SUBNET));
+    EXPECT_FALSE(URIUtils::IsOnLAN("127.0.0.1", ANY_PRIVATE_SUBNET));
+    EXPECT_FALSE(URIUtils::IsOnLAN("localhost", ANY_PRIVATE_SUBNET));
+
+    EXPECT_FALSE(URIUtils::IsOnLAN("some_unresolveable_hostname", ANY_PRIVATE_SUBNET));
+    EXPECT_FALSE(URIUtils::IsOnLAN("www.google.com", ANY_PRIVATE_SUBNET));
+    EXPECT_FALSE(URIUtils::IsOnLAN("www.aol.com", ANY_PRIVATE_SUBNET));
+    EXPECT_FALSE(URIUtils::IsOnLAN("168.219.34.129", ANY_PRIVATE_SUBNET));
+
+    EXPECT_FALSE(URIUtils::IsOnLAN("192.168.0.3", ANY_PRIVATE_SUBNET));
+    EXPECT_FALSE(URIUtils::IsOnLAN("172.16.3.2", ANY_PRIVATE_SUBNET));
+    EXPECT_FALSE(URIUtils::IsOnLAN("10.0.0.9", ANY_PRIVATE_SUBNET));
```

Comment:
Are those tests of any use? `URIUtils::IsOnLAN` is for URIs, it will always return `false` for `ANY_PRIVATE_SUBNET` if tested with invalid input. E.g. change `URIUtils::IsOnLAN("192.168.0.3", ANY_PRIVATE_SUBNET)` into `URIUtils::IsOnLAN("http://192.168.0.3", ANY_PRIVATE_SUBNET)` and it returns the expected `true`.